.circleci/config.yml

```
@@ -802,7 +800,55 @@ jobs:
           working_directory: ui/packages/consul-ui
           command: make test-coverage-ci
       - run: *notify-slack-failure
-
+      
+  upgrade-integration-test-latest:
```

could we call this something else? Like:

```suggestion
  testcontainer-integration-test:
```

Today we just test the `compatibility` use case but I can see this easily growing to encompass more use cases!

---

I don't want to tie this to testcontainer necessarily. I think we can also create other ways of instantiating Consul and run the exact same tests in different environments. I renamed it to `upgrade-integration-test` We can rename it again when we know what other use cases will fit.